src/interpreter/mod.rs

```diff
@@ -1404,6 +1404,10 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             (8, &ty::TyUint(UintTy::Us)) |
             (_, &ty::TyUint(UintTy::U64)) => PrimVal::U64(self.memory.read_uint(ptr, 8)? as u64),
 
+            (_, &ty::TyFnDef(def_id, substs, fn_ty)) => {
+                PrimVal::FnPtr(self.memory.create_fn_ptr(def_id, substs, fn_ty))
```

Comment:
`write_primval` is a shortcut for dumping the result of a mathematical operation into memory. We could rename it `write_numeric_primval` to make it clearer